packages/backend-defaults/src/entrypoints/database/connectors/postgres.ts

```diff
@@ -121,6 +138,119 @@ export async function buildPgDatabaseConfig(
   };
 }
 
+export type AzureTokenCredentialConfig = {
+  /**
+   * How early before an access token expires to refresh it with a new one.
+   * Defaults to 5 minutes
+   * Supported formats:
+   * - A string in the format of '1d', '2 seconds' etc. as supported by the `ms`
+   *   library.
+   * - A standard ISO formatted duration string, e.g. 'P2DT6H' or 'PT1M'.
+   * - An object with individual units (in plural) as keys, e.g. `{ days: 2, hours: 6 }`.
+   */
+  tokenRenewalOffsetTime?: string | HumanDuration;
+} & (
+  | {
+      type: 'ClientSecretCredential';
+      clientId: string;
+      clientSecret: string;
+      tenantId: string;
+    }
+  | {
+      type: 'ManagedIdentityCredential';
+      /**
+       * The client ID of a user-assigned managed identity.
+       * If not provided, the system-assigned managed identity is used.
+       */
+      clientId?: string;
+    }
+  | {
+      type: undefined | 'DefaultAzureCredential';
```

Comment:
```suggestion
      type?: undefined | 'DefaultAzureCredential';
```

needs to be truly optional right? 🤔